hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/CompactionHoodiePathCommand.scala

```diff
@@ -17,103 +17,70 @@
-             }
-           }
-        } else {
-          // Check if the compaction timestamp has exists in the pending compaction
-          if (pendingCompactionInstants.contains(instantTimestamp.get.toString)) {
-            Seq(instantTimestamp.get.toString)
-          } else {
-            throw new IllegalArgumentException(s"Compaction instant: ${instantTimestamp.get} is not found in $path," +
-              s" Available pending compaction instants are: ${pendingCompactionInstants.mkString(",")} ")
-          }
-        }
-        if (willCompactionInstants.isEmpty) {
-          logInfo(s"No need to compaction on $path")
-          Seq.empty[Row]
-        } else {
-          logInfo(s"Run compaction at instants: [${willCompactionInstants.mkString(",")}] on $path")
-          val timer = new HoodieTimer
-          timer.startTimer()
-          willCompactionInstants.foreach {compactionInstant =>
-            val writeResponse = client.compact(compactionInstant)
-            handleResponse(writeResponse.getCommitMetadata.get())
-            client.commitCompaction(compactionInstant, writeResponse.getCommitMetadata.get(), HOption.empty())
-          }
-          logInfo(s"Finish Run compaction at instants: [${willCompactionInstants.mkString(",")}]," +
-            s" spend: ${timer.endTimer()}ms")
-          Seq.empty[Row]
-        }
-      case _=> throw new UnsupportedOperationException(s"Unsupported compaction operation: $operation")
+    val op = operation match {
+      case SCHEDULE => UTF8String.fromString(Constants.SCHEDULE)
+      case RUN => UTF8String.fromString(Constants.RUN)
+      case _ => throw new UnsupportedOperationException(s"Unsupported compaction operation: $operation")
     }
-  }
 
-  private def handleResponse(metadata: HoodieCommitMetadata): Unit = {
-
-    // Handle error
-    val writeStats = metadata.getPartitionToWriteStats.entrySet().flatMap(e => e.getValue).toList
-    val errorsCount = writeStats.map(state => state.getTotalWriteErrors).sum
-    if (errorsCount > 0) {
-      throw new HoodieException(s" Found $errorsCount when writing record")
-    }
+    val procedureArgs = buildProcedureArgs(op, UTF8String.fromString(path), instantTimestamp)
+    RunCompactionProcedure.builder.get().build.call(procedureArgs)
   }
 
   override val output: Seq[Attribute] = {
     operation match {
       case RUN => Seq.empty
-      case SCHEDULE => Seq(AttributeReference("instant", StringType, nullable = false)())
+      case SCHEDULE => Seq(AttributeReference(Constants.INSTANT, StringType, nullable = false)())
     }
   }
+
+  /**
+   * Build run compaction procedure named args from given arguments
+   *
+   * @param op        The operation type, must be specified one of RUN or SCHEDULE
+   * @param tablePath The target table path
+   * @param timestamp The timestamp
+   * @return Run compaction procedure named args
+   */
+  private def buildProcedureArgs(op: UTF8String,
```

Comment:
`buildProcedureArgs` This can be extracted into a method to support both `CompactionShowHoodiePathCommand` and 
 `CompactionHoodiePathCommand`